packages/snaps-utils/src/iframe.ts

```diff
@@ -29,6 +29,10 @@ export async function createWindow(
     // MDN article for `load` event: https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event
     // Re: `load` firing twice: https://stackoverflow.com/questions/10781880/dynamically-created-iframe-triggers-onload-event-twice/15880489#15880489
     iframe.setAttribute('src', uri);
+
+    // Enable WebHID to be executed in the iframe.
+    iframe.setAttribute('allow', 'hid');
```

Comment:
Would be interesting if we could set this conditionally based on the permission 🤔

Reply:
The  snaps iframe is created before the snaps controller, so i'm not sure if we can conditionally set that.

Reply:
Hmm, it would be better if we didn't give HID to any iframes that don't need it, hmm 🤔

Reply:
The iframe is created by the execution service, after the snaps controller starts a snap, right? We could pass the endowments to the execution service, and only set this attribute based on that.

Reply:
Maarten is right, the iframe is created when `executeSnap` is called on an execution service. The `IframeExecutionService` does not create the iframe before that. So we can pass along the endowments from `snapData` and use that.

For reference:
https://github.com/MetaMask/snaps-monorepo/blob/main/packages/snaps-controllers/src/services/AbstractExecutionService.ts#L340

https://github.com/MetaMask/snaps-monorepo/blob/main/packages/snaps-controllers/src/services/AbstractExecutionService.ts#L220

https://github.com/MetaMask/snaps-monorepo/blob/main/packages/snaps-controllers/src/services/iframe/IframeExecutionService.ts#L40